### PR TITLE
Refactor Cassandra pools to use worker_pool

### PR DESCRIPTION
**Pull request by rslota**
This PR removes custom worker pools from Cassandra abstraction layer in favour of `worker_pool`.

TODO:
- [x] Change Cassandra read operation's callback function to be run in context of the original caller, not pool worker, since this can deadlock the system when read callbacks also call Cassandra

### CODE REVIEW

**Comment by michalwski** [comment]
@rslota could you point me where is the read operation and the callback you talk about in the TODO item?

**Comment by mongoose-im** [comment]
[5499.1](https://travis-ci.org/esl/MongooseIM/jobs/422558584) / Erlang 19.3 / small_tests / ba4de2ff1b9eb813178d86a9ee8743cba560221a
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2043/5499/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2043/5499/small_tests.19.3/small/ct_run.mongooseim@localhost.2018-08-30_15.50.33/lib.mongooseim.logs/run.2018-08-30_15.50.34/suite.log.html)

**Comment by mongoose-im** [comment]
[5530.3](https://travis-ci.org/esl/MongooseIM/jobs/424307124) / Erlang 19.3 / mysql_redis / 20f00ae4125087caf504a5c6e3e48e0a30efa2d6
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2043/5530/mysql_redis.19.3)/ [big](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2043/5530/mysql_redis.19.3/big//index.html)

**Comment by mongoose-im** [comment]
[5533.1](https://travis-ci.org/esl/MongooseIM/jobs/424470535) / Erlang 19.3 / small_tests / 4c4ea4e3d17003352398de874498cdd6ee54d72f
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2043/5533/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2043/5533/small_tests.19.3/small/ct_run.mongooseim@localhost.2018-09-04_19.04.02/lib.mongooseim.logs/run.2018-09-04_19.04.03/suite.log.html)

**Comment by mongoose-im** [comment]
[5549.1](https://travis-ci.org/esl/MongooseIM/jobs/425641064) / Erlang 19.3 / small_tests / f5308830dd55e8c8d2427e00e531093215c08384
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2043/5549/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2043/5549/small_tests.19.3/small/ct_run.mongooseim@localhost.2018-09-07_09.05.55/lib.mongooseim.logs/run.2018-09-07_09.05.56/suite.log.html)

**Comment by mongoose-im** [comment]
[5557.1](https://travis-ci.org/esl/MongooseIM/jobs/426557527) / Erlang 19.3 / small_tests / b8ce7aebe133b4761d32274864b1154edeba81f0
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2043/5557/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2043/5557/small_tests.19.3/small/ct_run.mongooseim@localhost.2018-09-10_07.38.34/lib.mongooseim.logs/run.2018-09-10_07.38.35/suite.log.html)

File: src/cassandra/mongoose_cassandra_pool.erl [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+call_query(PoolName, ContextId, Call) ->
+    mongoose_wpool:call(wpool_name(PoolName, query), global, default, ContextId, Call).
+
+cast_query(PoolName, undefined, Call) ->
+    mongoose_wpool:cast(wpool_name(PoolName, query), Call);
+cast_query(PoolName, ContextId, Call) ->
+    mongoose_wpool:cast(wpool_name(PoolName, query), global, default, ContextId, Call).
+
+wpool_name(PoolName, Type) ->
+    NameStr = atom_to_list(?MODULE) ++ "_" ++ atom_to_list(PoolName) ++ "_" ++ atom_to_list(Type),
```
Comment (kzemek) [inline src/cassandra/mongoose_cassandra_pool.erl]: `mongoose_wpool` pools are now named using `{Type :: atom(), Host :: binary() | global, Tag :: atom()}` tuple. A more idiomatic way to name a Cassandra pool right now would be `{?MODULE, global, PoolName}` (looks like `query` can be dropped).

**Comment by mongoose-im** [comment]
[5567.1](https://travis-ci.org/esl/MongooseIM/jobs/427128977) / Erlang 19.3 / small_tests / 6d1c798f3e10e63d9948672d40fa1ced23f29172
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2043/5567/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2043/5567/small_tests.19.3/small/ct_run.mongooseim@localhost.2018-09-11_10.57.31/lib.mongooseim.logs/run.2018-09-11_10.57.32/suite.log.html)

**Pull request merged**